### PR TITLE
backend.egl: Early return on dead buffers 

### PR DESCRIPTION
**Pull request by PolyMeilex**
fixes: #351

### CODE REVIEW

File: src/backend/egl/display.rs [hunk 1]
```diff
@@ -694,10 +698,16 @@ unsafe impl Send for EGLBufferReader {}
 
 #[cfg(feature = "use_system_lib")]
 impl EGLBufferReader {
-    fn new(display: Arc<EGLDisplayHandle>, wayland: *mut wl_display) -> Self {
+    fn new<L>(display: Arc<EGLDisplayHandle>, wayland: *mut wl_display, log: L) -> Self
+    where
+        L: Into<Option<::slog::Logger>>,
+    {
+        let logger = crate::slog_or_fallback(log.into()).new(o!("smithay_module" => "backend_egl"));
```
Comment (vberger) [inline src/backend/egl/display.rs:705]: Could you make the module as `"egl_buffer_reader"` instead, to be more specific?

**Review by vberger — CHANGES_REQUESTED** [review]
Ok, LGTM, just a minor nitpick, and could you please add a changelog entry for this bugfix as well?

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,6 +2,8 @@
 
 ## Unreleased
 
+- EGLBufferReader now checks if buffers are alive beffore using them.
```
Comment (vberger) [inline CHANGELOG.md:5]: There is a typo at "beffore", also could you make a `### Bugfixes` subsection including your change?

Comment (PolyMeilex) [reply CHANGELOG.md]: fixed, sorry about that.

**Review by vberger — APPROVED** [review]

**Pull request merged**